### ISSUE TITLE
Dismissing drawer tapping the black overlay calls onTap on ListTile if Talkback is on

### ISSUE THREAD

**Issue by lorenzOliveto** [flutter/flutter#39896]
# Steps to Reproduce
Create a basic Scaffold with a drawer and a ListView as a child with some ListTile elements, enable Talkback on Android and double tap on the black overlay to dismiss the drawer.
OnTap of a ListTile is called.
Here's a simple demo https://gist.github.com/lorenzOliveto/a8766a55a8eb4b726b874f9b0540c06f
Tried on a Nexus 5X.

## Flutter doctor output

```
flutter doctor -v
[✓] Flutter (Channel stable, v1.7.8+hotfix.4, on Mac OS X 10.14.6 18G95, locale it-IT)
    • Flutter version 1.7.8+hotfix.4 at /Users/lorenzooliveto/Flutter/flutter
    • Framework revision 20e59316b8 (7 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/lorenzooliveto/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (2 available)
    • iPhone di Lorenzo • 63e1defd8bca92bd059530106598ec853b3609ac • ios • iOS 13.1
    • iPhone Xs         • 58A5AE70-C88B-40F5-B64E-D3A884B4C30D     • ios • com.apple.CoreSimulator.SimRuntime.iOS-12-4 (simulator)

• No issues found!
```

**Comment by perclasson** [comment]
From https://github.com/flutter/flutter/issues/1715 https://github.com/flutter/flutter/pull/11024 it seems as we should not be able to dismiss the drawer with a modal on Android, and instead we should use the back button. So by simply excluding the overlay on Android we should be able to fix this.
